test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/HeaderModelBinderTests.cs

```diff
@@ -102,36 +107,105 @@ public class HeaderModelBinderTests
         }
 
         [Fact]
-        public async Task HeaderBinder_ReturnsResult_ForReadOnlyDestination()
+        public async Task HeaderBinder_BindsHeaders_ToStringCollection()
         {
             // Arrange
-            var header = "Accept";
+            var type = typeof(string[]);
             var headerValue = "application/json,text/json";
-            var binder = new HeaderModelBinder(NullLoggerFactory.Instance);
-            var bindingContext = GetBindingContextForReadOnlyArray();
+            var bindingContext = CreateContext(type);
+            var binder = CreateBinder(bindingContext);
+            bindingContext.HttpContext.Request.Headers.Add("Header", new[] { headerValue });
 
-            bindingContext.FieldName = header;
-            bindingContext.HttpContext.Request.Headers.Add(header, new[] { headerValue });
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            Assert.Equal(headerValue.Split(','), bindingContext.Result.Model);
+        }
+
+        public static TheoryData<string, Type, object> BinderHeaderToSimpleTypesData
+        {
+            get
+            {
+                var guid = Guid.NewGuid();
```

Comment:
Can you use fixed `Guid` here? Having varying data throws off our CI's test collection (since it gets reported as a different test name for each run

Reply:
Ah yes! Good catch